### PR TITLE
Refactor runtime metrics to use localhost accelerated table

### PR DESCRIPTION
**Pull request by ewgenius**
- Added `InternalTable`  wrapper around `AcceleratedTable` for internal spice metrics.
- Fixed localhost data connector, to return `EmptyExec` on insert.
- Added default retention settings for `runtime.metrics` to delete records older than 30 minutes every 5 minutes.

### CODE REVIEW

File: crates/runtime/src/internal_table.rs [hunk 1]
```diff
@@ -0,0 +1,140 @@
...
+        Ok(())
+    }
+
+    pub async fn new(
+        name: &str,
+        schema: Arc<Schema>,
+        acceleration: Acceleration,
+        refresh: Refresh,
+        retention: Option<Retention>,
+    ) -> Result<Self, Error> {
```
Comment (ewgenius) [inline crates/runtime/src/internal_table.rs]: Only localhost connector for now. As a next step I will add option to initialize internal table with spiceai connection

File: crates/runtime/src/dataconnector/localhost.rs [hunk 2]
```diff
@@ -215,48 +217,9 @@ impl TableProvider for LocalhostConnector {
...
-        input: Arc<dyn ExecutionPlan>,
+        _input: Arc<dyn ExecutionPlan>,
         _overwrite: bool,
     ) -> datafusion::error::Result<Arc<dyn ExecutionPlan>> {
-        Ok(Arc::new(DataSinkExec::new(
-            input,
-            Arc::new(LocalhostDataSink::new()),
-            self.schema(),
-            None,
-        )) as _)
```
Comment (ewgenius) [inline crates/runtime/src/dataconnector/localhost.rs:226]: That doesn't work well together with acceleration

File: bin/spiced/src/lib.rs [hunk 3]
```diff
@@ -104,16 +104,22 @@ pub async fn run(args: Args) -> Result<()> {
...
 
     rt.load_models().await;
 
+    if args.spice_cloud_connect {
+        if let Err(err) = rt
+            .start_metrics(args.metrics)
+            .await
+            .context(UnableToStartServersSnafu)
+        {
+            tracing::warn!("{}", err);
```
Comment (phillipleblanc) [inline bin/spiced/src/lib.rs]:
```suggestion
            tracing::warn!("{err}");
```

File: crates/runtime/src/spice_metrics.rs [hunk 4]
```diff
@@ -32,38 +39,58 @@ pub enum Error {
...
-            table: Arc::new(table),
-        }
+            metrics_table: Arc::new(internal_table),
+        })
     }
 
-    async fn tick(socket_addr: &SocketAddr, table: &Arc<MetricsTable>) -> Result<(), Error> {
+    async fn tick(
+        socket_addr: &SocketAddr,
+        data_fusion: &Arc<RwLock<DataFusion>>,
```
Comment (phillipleblanc) [inline crates/runtime/src/spice_metrics.rs]:
```suggestion
        datafusion: &Arc<RwLock<DataFusion>>,
```

datafusion is one word

**Review by digadeesh — APPROVED** [review]

**Review by phillipleblanc — APPROVED** [review]

**Pull request merged**